src/components/views/LocalModList/DisableModModal.vue

```diff
@@ -1,41 +1,75 @@
 <script lang="ts">
-import { Vue, Component, Prop } from 'vue-property-decorator';
+import { Vue, Component } from 'vue-property-decorator';
 import { ModalCard } from '../../all';
+import R2Error from '../../../model/errors/R2Error';
 import ManifestV2 from '../../../model/ManifestV2';
+import LoggerProvider, { LogSeverity } from '../../../providers/ror2/logging/LoggerProvider';
+import Dependants from '../../../r2mm/mods/Dependants';
 
 @Component({
     components: {ModalCard}
 })
 export default class DisableModModal extends Vue {
 
-    @Prop({required: true})
-    readonly mod!: ManifestV2;
+    modBeingDisabled: string | null = null;
 
-    @Prop({required: true})
-    readonly dependencyList!: Set<ManifestV2>;
+    get dependants() {
+        return Dependants.getDependantList(this.mod, this.$store.state.localModList);
+    }
+
+    get isLocked(): boolean {
+        return this.modBeingDisabled !== null;
+    }
+
+    get isOpen(): boolean {
+        return this.$store.state.modals.isDisableModModalOpen
+            && this.$store.state.modals.disableModModalMod !== null;
+    }
 
-    @Prop({required: true})
-    readonly dependantsList!: Set<ManifestV2>;
+    get mod(): ManifestV2 {
+        if (this.$store.state.modals.disableModModalMod === null) {
+            this.$emit('error', new R2Error(
+                'Error while opening DisableModModal',
+                'Mod not provided'
+            ));
```

Comment:
This should probably throw instead of just emit an error because this is also relied on functions such as `disableModIncludingDependants` which should fail if this fails. I'm not sure if whether or not that scenario can actually occur right now but it seems like something that could happen due to a race condition given the async context we're dealing with.

Reply:
It's intended as a "this should never happen" safety, since the whole component is mounted only if `isOpen` is true, which checks that the mod is actually provided. That being said, I guess throwing might be marginally better approach.

Changed.